app/tests/test_middlewares.py

```diff
@@ -16,23 +18,32 @@
             {"user_id": "test_user"},
             lambda: "Test Response",
             b"Test Response",
-            f'{{"event": "api_request", "user_id": "test_user", "route": "{HOST}/test_route", "method": "GET"}}',
+            json.loads(
+                """{"event": "api_request", "user_id": "test_user", "route": "http://localhost/test_route",
+                "method": "GET", "caller_function": "dynamic_route", "caller_module": "app.tests.test_middlewares"}"""
+            ),
         ),
         (
             "/anonymous_route",
             "GET",
             {},
             lambda: "Anonymous Response",
             b"Anonymous Response",
-            f'{{"event": "api_request", "user_id": "anonymous", "route": "{HOST}/anonymous_route", "method": "GET"}}',
+            json.loads(
+                """{"event": "api_request", "user_id": "anonymous", "route": "http://localhost/anonymous_route",
+                "method": "GET", "caller_function": "dynamic_route", "caller_module": "app.tests.test_middlewares"}"""
+            ),
         ),
         (
             "/post_route",
             "POST",
             {"user_id": "test_user"},
             lambda: "Post Response",
             b"Post Response",
-            f'{{"event": "api_request", "user_id": "test_user", "route": "{HOST}/post_route", "method": "POST"}}',
+            json.loads(
```

Comment:
why json.loads vs just defining the dict like before?

Reply:
Either way works, though it would be good to load the dict directly

Reply:
cool